// @fluentui/react-button/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Button: function() {
        return _Button.Button;
    },
    ButtonContextProvider: function() {
        return _index1.ButtonContextProvider;
    },
    CompoundButton: function() {
        return _CompoundButton.CompoundButton;
    },
    MenuButton: function() {
        return _MenuButton.MenuButton;
    },
    SplitButton: function() {
        return _SplitButton.SplitButton;
    },
    ToggleButton: function() {
        return _ToggleButton.ToggleButton;
    },
    buttonClassNames: function() {
        return _Button.buttonClassNames;
    },
    compoundButtonClassNames: function() {
        return _CompoundButton.compoundButtonClassNames;
    },
    menuButtonClassNames: function() {
        return _MenuButton.menuButtonClassNames;
    },
    renderButton_unstable: function() {
        return _Button.renderButton_unstable;
    },
    renderCompoundButton_unstable: function() {
        return _CompoundButton.renderCompoundButton_unstable;
    },
    renderMenuButton_unstable: function() {
        return _MenuButton.renderMenuButton_unstable;
    },
    renderSplitButton_unstable: function() {
        return _SplitButton.renderSplitButton_unstable;
    },
    renderToggleButton_unstable: function() {
        return _ToggleButton.renderToggleButton_unstable;
    },
    splitButtonClassNames: function() {
        return _SplitButton.splitButtonClassNames;
    },
    toggleButtonClassNames: function() {
        return _ToggleButton.toggleButtonClassNames;
    },
    useButtonContext: function() {
        return _index1.useButtonContext;
    },
    useButtonStyles_unstable: function() {
        return _Button.useButtonStyles_unstable;
    },
    useButton_unstable: function() {
        return _Button.useButton_unstable;
    },
    useCompoundButtonStyles_unstable: function() {
        return _CompoundButton.useCompoundButtonStyles_unstable;
    },
    useCompoundButton_unstable: function() {
        return _CompoundButton.useCompoundButton_unstable;
    },
    useMenuButtonStyles_unstable: function() {
        return _MenuButton.useMenuButtonStyles_unstable;
    },
    useMenuButton_unstable: function() {
        return _MenuButton.useMenuButton_unstable;
    },
    useSplitButtonStyles_unstable: function() {
        return _SplitButton.useSplitButtonStyles_unstable;
    },
    useSplitButton_unstable: function() {
        return _SplitButton.useSplitButton_unstable;
    },
    useToggleButtonStyles_unstable: function() {
        return _ToggleButton.useToggleButtonStyles_unstable;
    },
    useToggleButton_unstable: function() {
        return _ToggleButton.useToggleButton_unstable;
    },
    useToggleState: function() {
        return _index.useToggleState;
    }
});
const _Button = require("./Button");
const _CompoundButton = require("./CompoundButton");
const _MenuButton = require("./MenuButton");
const _SplitButton = require("./SplitButton");
const _ToggleButton = require("./ToggleButton");
const _index = require("./utils/index");
const _index1 = require("./contexts/index");
 // Experimental APIs - will be uncommented in the experimental release branch
 // export { useButtonBase_unstable } from './Button';
 // export type { ButtonBaseProps, ButtonBaseState } from './Button';
 // export { useCompoundButtonBase_unstable } from './CompoundButton';
 // export type { CompoundButtonBaseProps, CompoundButtonBaseState } from './CompoundButton';
 // export { useMenuButtonBase_unstable } from './MenuButton';
 // export type { MenuButtonBaseProps, MenuButtonBaseState } from './MenuButton';
 // export { useSplitButtonBase_unstable } from './SplitButton';
 // export type { SplitButtonBaseProps, SplitButtonBaseState } from './SplitButton';
 // export { useToggleButtonBase_unstable } from './ToggleButton';
 // export type { ToggleButtonBaseProps, ToggleButtonBaseState } from './ToggleButton';
